src/kubernetes_cleanup_jobs/kubernetes_cleanup_jobs/config.py

```diff
@@ -0,0 +1,89 @@
+import os
+import argparse
+import logging
+
+import kubernetes_cleanup_jobs.context
+from lib.logging import setup_logging
+
+class AppConfig:
+  def __init__(self, cmdline_args, env):
+    ''' set app-wide configs and initialize the app '''
+    nhours, name, completed, all, namespace, force, loglevel = (
+      int(cmdline_args.nhours),
+      cmdline_args.name,
+      cmdline_args.completed,
+      cmdline_args.all,
+      cmdline_args.namespace,
+      cmdline_args.force,
+      cmdline_args.loglevel,
+    )
+
+    context = env
+
+    self.nhours = nhours
+    self.name = name
+    self.completed = completed
+    self.all = all
+    self.namespace = namespace
+    self.force = force
+    self.context = context
+
+    self._init_app(loglevel)
+
+  def _init_app(self, loglevel):
+    ''' initialize the app '''
+    setup_logging(eval('logging.' + loglevel))
+
+def parse_args():
+  ''' parse command line args '''
+  parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter
+  )
+  parser.add_argument(
+    'nhours',
+    help='delete jobs older than n hours'
+  )
+  group = parser.add_mutually_exclusive_group(required=True)
+  group.add_argument(
+    '--name',
+    help='delete jobs by name'
+  )
+  group.add_argument(
+    '--completed',
+    action='store_true',
+    help='delete completed jobs'
+  )
+  group.add_argument(
+    '--all',
```

Comment:
I decided to implement the `--all` option instead of continuing to use `--incomplete` to trigger a cleanup of all jobs, including those that have not successfully completed.